libs/qCC_io/LASFilter.cpp

```diff
@@ -32,1408 +32,1167 @@
-			if (w)
-			{
-				w->WritePoint(P);
-				++writeCounter;
-			}
-		}
-
-		void close()
-		{
-			if (w)
-			{
-				delete w;
-				w = 0;
-				ofs.close();
-			}
-		}
-
-		inline size_t writtenCount() const { return writeCounter; }
-
-		const QString& getFilename() const { return filename; };
-
-protected:
+	switch (sfType) {
+	case LAS_FIELDS::LAS_X:
+		return pdal::Dimension::Id::X;
+	case LAS_FIELDS::LAS_Y:
+		return pdal::Dimension::Id::Y;
+	case LAS_FIELDS::LAS_Z:
+		return pdal::Dimension::Id::Z;
+	case LAS_FIELDS::LAS_INTENSITY:
+		return pdal::Dimension::Id::Intensity;
+	case LAS_FIELDS::LAS_RETURN_NUMBER:
+		return pdal::Dimension::Id::ReturnNumber;
+	case LAS_FIELDS::LAS_NUMBER_OF_RETURNS:
+		return pdal::Dimension::Id::NumberOfReturns;
+	case LAS_FIELDS::LAS_SCAN_DIRECTION:
+		return pdal::Dimension::Id::ScanDirectionFlag;
+	case LAS_FIELDS::LAS_FLIGHT_LINE_EDGE:
+		return pdal::Dimension::Id::EdgeOfFlightLine;
+	case LAS_FIELDS::LAS_CLASSIFICATION:
+		return  pdal::Dimension::Id::Classification;
+	case LAS_FIELDS::LAS_SCAN_ANGLE_RANK:
+		return pdal::Dimension::Id::ScanAngleRank;
+	case LAS_FIELDS::LAS_USER_DATA:
+		return pdal::Dimension::Id::UserData;
+	case LAS_FIELDS::LAS_POINT_SOURCE_ID:
+		return pdal::Dimension::Id::PointSourceId;
+	case LAS_FIELDS::LAS_RED:
+		return pdal::Dimension::Id::Red;
+	case LAS_FIELDS::LAS_GREEN:
+		return pdal::Dimension::Id::Green;
+	case LAS_FIELDS::LAS_BLUE:
+		return pdal::Dimension::Id::Blue;
+	case LAS_FIELDS::LAS_TIME:
+		return pdal::Dimension::Id::GpsTime;
+	case LAS_FIELDS::LAS_EXTRA:
+		return pdal::Dimension::Id::Unknown;
+		//Sub fields
+	case LAS_FIELDS::LAS_CLASSIF_VALUE:
+		return pdal::Dimension::Id::Classification;
+	case LAS_FIELDS::LAS_CLASSIF_SYNTHETIC:
+		return pdal::Dimension::Id::ClassFlags;
+	case LAS_FIELDS::LAS_CLASSIF_KEYPOINT:
+		return pdal::Dimension::Id::ClassFlags;
+	case LAS_FIELDS::LAS_CLASSIF_WITHHELD:
+		return pdal::Dimension::Id::ClassFlags;
+		//Invald flag
+	case LAS_FIELDS::LAS_INVALID:
+	default:
+		return pdal::Dimension::Id::Unknown;
+	};
+}
 
-	liblas::Writer* w;
-	std::ofstream ofs;
-	QString filename;
-	size_t writeCounter;
-};
 
 CC_FILE_ERROR LASFilter::saveToFile(ccHObject* entity, QString filename, SaveParameters& parameters)
 {
-	if (!entity || filename.isEmpty())
-		return CC_FERR_BAD_ARGUMENT;
-
-	ccGenericPointCloud* theCloud = ccHObjectCaster::ToGenericPointCloud(entity);
-	if (!theCloud)
-	{
-		ccLog::Warning("[LAS] This filter can only save one cloud at a time!");
-		return CC_FERR_BAD_ENTITY_TYPE;
-	}
-
-	unsigned numberOfPoints = theCloud->size();
-	if (numberOfPoints == 0)
+    if (!entity || filename.isEmpty())
+        return CC_FERR_BAD_ARGUMENT;
+
+    ccGenericPointCloud* theCloud = ccHObjectCaster::ToGenericPointCloud(entity);
+    if (!theCloud)
+    {
+        ccLog::Warning("[LAS] This filter can only save one cloud at a time");
+        return CC_FERR_BAD_ENTITY_TYPE;
+    }
+
+    unsigned numberOfPoints = theCloud->size();
```

Comment:
So this is just my opinion - I don't think our brief style guide has anything to say about it - but I prefer `unsigned int` to simply `unsigned`. Whichever is chosen, please use consistently throughout.